ddl/lightning/engine_mgr.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lightning
+
+import (
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/util/generic"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+type engineManager struct {
+	generic.SyncMap[int64, *engineInfo]
+	MemRoot  MemRoot
+	DiskRoot DiskRoot
+}
+
+func (m *engineManager) init(memRoot MemRoot, diskRoot DiskRoot) {
+	m.SyncMap = generic.NewSyncMap[int64, *engineInfo](10)
+	m.MemRoot = memRoot
+	m.DiskRoot = diskRoot
+}
+
+// Register create a new engineInfo and register it to the engineManager.
+func (m *engineManager) Register(bc *BackendContext, job *model.Job, indexID int64) error {
+	// Calculate lightning concurrency degree and set memory usage
+	// and pre-allocate memory usage for worker.
+	m.MemRoot.RefreshConsumption()
+	ok := m.MemRoot.TestConsume(int64(bc.cfg.TikvImporter.LocalWriterMemCacheSize))
```

Comment:
Maybe we can rename this function `TestConsume`, like `checkConsume` or other names. The name confused me a little bit, thinking it was code for testing.

Reply:
It is suggested by @xiongjiwei ... let me change it to `CheckConsume`.